core/src/main/scala/quasar/destination/redshift/RedshiftDestinationModule.scala

```diff
@@ -1,8 +1,179 @@
+/*
+ * Copyright 2014–2019 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package quasar.destination.redshift
 
-import quasar.api.destination.Destination
-import quasar.connector.DestinationModule
+import scala._
+import scala.Predef._
+import scala.concurrent.ExecutionContext
+import scala.util.Random
+
+import quasar.api.destination.DestinationError.InitializationError
+import quasar.api.destination.{Destination, DestinationError, DestinationType}
+import quasar.blobstore.s3.{
+  AccessKey,
+  Bucket,
+  Region,
+  S3DeleteService,
+  S3PutService,
+  S3StatusService,
+  SecretKey
+}
+import quasar.blobstore.BlobstoreStatus
+import quasar.connector.{DestinationModule, MonadResourceErr}
+import quasar.concurrent.NamedDaemonThreadFactory
+
+import argonaut._, Argonaut._
+
+import cats.effect.{ConcurrentEffect, Concurrent, ContextShift, Resource, Sync, Timer}
+import cats.data.EitherT
+import cats.implicits._
+
+import doobie.Transactor
+import doobie.free.connection.isValid
+import doobie.hikari.HikariTransactor
+import doobie.implicits._
+
+import eu.timepit.refined.auto._
+
+import java.util.concurrent.Executors
+
+import scalaz.NonEmptyList
+
+import software.amazon.awssdk.auth.credentials.{AwsBasicCredentials, StaticCredentialsProvider}
+import software.amazon.awssdk.regions.{Region => AwsRegion}
+import software.amazon.awssdk.services.s3.S3AsyncClient
+
+
+object RedshiftDestinationModule extends DestinationModule {
+  val RedshiftDriverFqcn = "com.amazon.redshift.jdbc.Driver"
+  val PoolSize = 10
```

Comment:
Maybe this and could be provided in config with defaulting value `10`? Or calculated based on cores number?

Reply:
This is a connection pools so it's IO-bound rather than CPU-bound, so I don't think it makes sense to make it depend on CPUs. 

We could make it configurable through the destination config, but not sure if worthwhile. Ideally, quasar would handle the connection pools and they'd be configurable through SDBE rather than through the destination.